File: JavaScript/문제_24479번.js
let input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
for (let i = 0; i < input.length; i++) {
    input[i] = input[i].split(' ');
}
let n = +input[0][0];
let m = +input[0][1];
let r = +input[0][2];
let node = new Array(n);
const visit = new Array(n).fill(0);

for (let i = 0; i < n; i++) {
    node[i] = [];
}

for (let i = 1; i <= m; i++) {
    let from = +input[i][0];
    let to = +input[i][1];
    node[from - 1].push(to - 1);
    node[to - 1].push(from - 1);
}

for (let i = 0; i < n; i++) {
    node[i].sort((o1, o2) => {
        return o1 - o2;
    });
}
let cnt = +1;
let dfs = (start) => {
    visit[start] = cnt++;
    for (let y of node[start]) {
        if (visit[y] == 0) {
            dfs(y);
        }
    }
}
dfs(r - 1);
console.log(visit.join('\n'));

File: C++/문제_24479번.cpp
#include <iostream>
#include <algorithm>
using namespace std;

//문제 24479번 알고리즘 수업 - 깊이 우선 탐색 1 -> DFS
#include <vector>
int visit[100001];
int cnt = 1;
void dfs(vector<int>v[], int now){
    visit[now] = cnt++;
    for(int next : v[now]){
        if(visit[next] == 0){
            dfs(v, next);
        }
    }
}
int main(){
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);
    cout.tie(NULL);
    vector<int>v[100001];
    int n, m, r;
    cin >> n >> m >> r;
    for(int i = 0;i<m;i++){
        int from, to;
        cin >> from >> to;
        v[from].push_back(to);
        v[to].push_back(from);
    }
    for(int i = 0;i<=n;i++){
        sort(v[i].begin(), v[i].end());
    }
    dfs(v, r);
    for(int i = 1;i<=n;i++){
        if(visit[i] == 0)
            cout << "0" << "\n";
        else
            cout << visit[i] << "\n";
    }
}